airflow-core/src/airflow/api_fastapi/core_api/routes/public/hitl.py

```diff
@@ -272,3 +293,456 @@ def get_hitl_details(
+    session: SessionDep,
+) -> HITLDetailResponse:
+    """
+    Create a shared link for a mapped Human-in-the-loop task.
+
+    This endpoint generates a secure, time-limited shared link for mapped task instances,
+    allowing external users to interact with specific mapped HITL tasks without requiring
+    full Airflow authentication. The link can be configured for either direct action
+    execution or UI redirection.
+
+    :param dag_id: The DAG identifier
+    :param dag_run_id: The DAG run identifier
+    :param task_id: The task identifier
+    :param map_index: The map index for the mapped task instance
+    :param update_hitl_detail_payload: Payload containing link configuration and initial response data
+    :param user: The authenticated user creating the shared link
+    :param session: Database session for data persistence
+    """
+    if not hitl_shared_link_manager.is_enabled():
+        raise HTTPException(
+            status.HTTP_403_FORBIDDEN,
+            "HITL shared links are not enabled",
+        )
+
+    task_instance = _get_task_instance(
+        dag_id=dag_id,
+        dag_run_id=dag_run_id,
+        task_id=task_id,
+        session=session,
+        map_index=map_index,
+    )
+
+    ti_id_str = str(task_instance.id)
+    hitl_detail_model = session.scalar(select(HITLDetailModel).where(HITLDetailModel.ti_id == ti_id_str))
+    if not hitl_detail_model:
+        raise HTTPException(
+            status.HTTP_404_NOT_FOUND,
+            f"Human-in-the-loop detail does not exist for Task Instance with id {ti_id_str}",
+        )
+
+    try:
+        link_data = hitl_shared_link_manager.generate_link(
+            dag_id=dag_id,
+            dag_run_id=dag_run_id,
+            task_id=task_id,
+            map_index=map_index,
+            link_type=update_hitl_detail_payload.link_type,
+            action=update_hitl_detail_payload.action,
+            expires_in_hours=update_hitl_detail_payload.expires_in_hours,
+        )
+
+        response = HITLDetailResponse(
+            user_id=user.get_id(),
+            response_at=timezone.utcnow(),
+            chosen_options=update_hitl_detail_payload.chosen_options,
+            params_input=update_hitl_detail_payload.params_input,
+            task_instance_id=link_data["task_instance_id"],
+            link_url=link_data["link_url"],
+            expires_at=link_data["expires_at"],
+            action=link_data["action"],
+            link_type=link_data["link_type"],
+        )
+
+        return response
+
+    except ValueError as e:
+        raise HTTPException(
+            status.HTTP_400_BAD_REQUEST,
+            str(e),
+        )
+
+
+@hitl_router.get(
+    "/api/v2/hitl-details-share-link/{dag_id}/{dag_run_id}/{task_id}",
+    status_code=status.HTTP_200_OK,
+    responses=create_openapi_http_exception_doc(
+        [
+            status.HTTP_404_NOT_FOUND,
+            status.HTTP_400_BAD_REQUEST,
+            status.HTTP_403_FORBIDDEN,
+        ]
+    ),
```

Comment:
```suggestion
    responses=create_openapi_http_exception_doc(
        [
            status.HTTP_400_BAD_REQUEST,
            status.HTTP_403_FORBIDDEN,
            status.HTTP_404_NOT_FOUND,
        ]
    ),
```

The 200 status code is default by FastAPI itself.

Reply:
changed it